DCO Action

ehazlett opened the pull request:
This creates a GitHub action using the [dco](https://github.com/moby/moby/blob/master/hack/validate/dco) script from moby.  I didn't use the existing DCO check from [Probot](https://probot.github.io/) since it was third party and we already have an "approved" DCO script but I'm happy to use that instead if we don't need the custom validation.

I'm also happy to update our Poule setup if we don't want to use actions at this point either.  This just took _way_ less time to setup and seems easier to maintain :) 🤗

ehazlett commented:
Example PR with a DCO fail: https://github.com/ehazlett/buildkit/pull/1/checks

ehazlett commented:
I'll rebase once this is reviewed to show the DCO check succeed :)

tonistiigi commented:
Are you sure this gets called on PRs atm when only push event is allowed on public repos? I think your example only ran because you pushed the branch to the same repo. Tried to verify this in https://github.com/tiborvass/buildkit/pull/1 and it didn't trigger a check.

ehazlett commented:
Hmm good point.  I'm not sure if it's a "bootstrapping" thing that once it's enabled it works or ya if it's just for the main repo.  I'll do some quick digging.

ehazlett commented:
OK I updated the action to be `pull_request` instead of push.  According to the [docs](https://developer.github.com/actions/creating-workflows/workflow-configuration-options/#pull-request-events-for-forked-repositories) this should get executed whenever the PR is opened, updated, etc.  It will also only run the action on the base repo as this is where the action is enabled.

tonistiigi commented:
@ehazlett I think only `push` event is allowed on public repos during the beta period.

https://developer.github.com/actions/creating-workflows/workflow-configuration-options/#workflow-configuration-options

> Creating workflows that use GitHub Actions is limited to private repositories and push events in public repositories during the limited public beta.

ehazlett commented:
Ok I mis-read that.  I'll do a quick test repo and if it doesn't work we can go back to the Poule mechanism.

ehazlett commented:
OK I'll revisit this once Actions are opened to include `pull_request` on public repos.  For now, Poule has been enabled (see https://github.com/moby/buildkit/pull/793 for a test PR).
